_posts/2020-07-03-ansible-jak-uporzadkowac-chaos.md

```diff
@@ -0,0 +1,314 @@
+
+Ansible wymaga hasła lub klucza SSH w celu rozpoczęcia zarządzania systemami i może rozpocząć zarządzanie nimi bez instalowania oprogramowania agenta, unikając problemu „zarządzania zarządzaniem” powszechnego w wielu systemach do automatyzacji.
+
+Ansible łączy się z systemami poprzez mechanizmy transportowe – SSH (Unix) lub PowerShell (Windows). **Moduły**, które są małymi programami, zawierające uzupełnione argumenty, przenoszone są do tymczasowego katalogu przez wspomniane mechanizmy na zarządzane maszyny. Tam są wykonywane, a następnie usuwane w ramach jednej akcji. Moduły zwracają obiekty JSON na standardowe wyjście, a te z kolei przetwarzane są przez program Ansible na maszynie sterującej. Mogą one zarządzać idempotentnymi zasobami. Oznacza to, że moduł działa w sposób deklaratywny, czyli może zdecydować np. czy dany pakiet powinien zostać zainstalowany w określonej wersji lub nie wykonać żadnej akcji, gdy system jest już w pożądanym stanie. Można je też uruchamiać pojedynczo (imperatywnie).
+
+Do zarządzania administrowanymi systemami służą pliki **inventory**. Pozwalają one na grupowanie serwerów i definiowanie zmiennych, które później wykorzystywane będą w tzw. **playbookach**. Plik inventory może być użyty globalnie, gdzie zainstalowano Ansible lub wykorzystać lokalne pliki inventory dedykowane dla konkretnego projektu. Przykładowy fragment pliku inventory wygląda następująco:
+```yaml
+# eximee-ansible/settings/settings-template.yml
+
+### Ogólne parametry instalacji
+
+validation_protocol: http
+validation_host: eximee-validation
+validation_port: 8080
+services_protocol: http
+services_host: eximee-services
+services_port: 8080
+servicemix_host: eximee-servicemix
+{% raw %}validation_url: "{{ validation_protocol }}://{{ validation_host }}:{{ validation_port }}/osgi-bridge/validation/"
+services_url: "{{ services_protocol }}://{{ services_host }}:{{ services_port }}/osgi-bridge/serviceproxy/"
+eximee_status_url: "{{ eximee_status_protocol }}://{{ eximee_status_host }}:{{ eximee_status_port }}/eximee-status/"
+repository_url: "{{ repository_protocol }}://{{ repository_host }}:{{ repository_port }}/repository/"{% endraw %}
+repository_username: UZUPELNIJ
+repository_password: UZUPELNIJ
+mongodb_host: eximee-mongo
+mongo_db_auth: MONGODB-CR
+...
+```
+W pliku inventory znajdują się zatem dane dotyczące hostów, używanych portów czy dane uwierzytelniające. W inventory warto umieścić również takie zmienne, które są częściej wykorzystywane w playbookach lub zależne są np. od serwerów, na których instalujemy aplikację. Nie powinno się bowiem modyfikować playbooków przy okazji każdej instalacji, a utrzymywać dla danego serwera odpowiednią konfigurację. W przypadku platformy Eximee takimi zmiennymi są np. adresy URL różnych aplikacji. Warto też zwrócić uwagę, że adresy w powyższym przykładzie nie są zdefiniowane „na sztywno”. Zawierają one odwołania do kilku innych zmiennych tworząc nową wartość, na co również pozwala Ansible.
```

Comment:
właściwie podawanie danych uwierzytelniających jawnym tekstem w inventory to nie jest dobry wzorzec, w tym celu wykorzystuje się Ansible Vault

Reply:
słusznie! dodałem sekcję o vault